<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QWidget" name="SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>819</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SnugQt - Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="2" column="0">
    <widget class="QPushButton" name="saveSettingsButton">
     <property name="toolTip">
      <string>Save the current settings</string>
     </property>
     <property name="text">
      <string>Save settings</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QTabWidget" name="settingsTab">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="parametersTab">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>Parameters</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="1" column="1" colspan="2">
        <widget class="QComboBox" name="checkpointCombo"/>
       </item>
       <item row="41" column="0" colspan="3">
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Width:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="useExternalVaeCheck">
         <property name="toolTip">
          <string>Use an external VAE instead of model's built-in VAE</string>
         </property>
         <property name="text">
          <string>Use external VAE:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Height:</string>
         </property>
        </widget>
       </item>
       <item row="23" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Scheduler:</string>
         </property>
        </widget>
       </item>
       <item row="31" column="1">
        <widget class="QSpinBox" name="sdxlRefinerStepsValue">
         <property name="toolTip">
          <string>Number of steps to perform with SDXL refiner</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="value">
          <number>5</number>
         </property>
        </widget>
       </item>
       <item row="29" column="0">
        <widget class="QCheckBox" name="sdxlRefinerCheck">
         <property name="toolTip">
          <string>Enable refiner when using SDXL</string>
         </property>
         <property name="text">
          <string>SDXL Refiner</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="Line" name="line_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="42" column="0">
        <widget class="QCheckBox" name="modelUpscaleCheck">
         <property name="toolTip">
          <string>Upscale final image with upscale model</string>
         </property>
         <property name="text">
          <string>Upscale with model:</string>
         </property>
        </widget>
       </item>
       <item row="19" column="1">
        <widget class="QSpinBox" name="stepsValue">
         <property name="toolTip">
          <string>Number of steps to perform when generating iamge</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>400</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="42" column="1" colspan="2">
        <widget class="QComboBox" name="modelUpscaleCombo">
         <property name="toolTip">
          <string>Upscale model to use</string>
         </property>
        </widget>
       </item>
       <item row="24" column="1">
        <widget class="QSpinBox" name="clipSkipValue">
         <property name="toolTip">
          <string>Set last layer of CLIP. -1 for normal</string>
         </property>
         <property name="minimum">
          <number>-8</number>
         </property>
         <property name="maximum">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item row="31" column="0">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>SDXL refiner steps:</string>
         </property>
        </widget>
       </item>
       <item row="30" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>SDXL Refiner model:</string>
         </property>
        </widget>
       </item>
       <item row="16" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Iterations:</string>
         </property>
        </widget>
       </item>
       <item row="24" column="0">
        <widget class="QLabel" name="label_25">
         <property name="text">
          <string>Clip skip:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1" colspan="2">
        <widget class="QComboBox" name="externalVaeCombo"/>
       </item>
       <item row="28" column="0" colspan="3">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="23" column="1" colspan="2">
        <widget class="QComboBox" name="schedulerValue">
         <property name="toolTip">
          <string>Noise scheduler to use</string>
         </property>
         <item>
          <property name="text">
           <string>normal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>karras</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>exponential</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>simple</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>ddim_uniform</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="imgWidthValue">
         <property name="toolTip">
          <string>Width of the generated image</string>
         </property>
         <property name="minimum">
          <number>256</number>
         </property>
         <property name="maximum">
          <number>4096</number>
         </property>
         <property name="singleStep">
          <number>64</number>
         </property>
         <property name="value">
          <number>512</number>
         </property>
        </widget>
       </item>
       <item row="22" column="1" colspan="2">
        <widget class="QComboBox" name="samplerValue">
         <property name="toolTip">
          <string>Noise sampler to use</string>
         </property>
         <item>
          <property name="text">
           <string>euler</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>euler_ancestral</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>heun</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpm_2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpm_2_ancestral</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>lms</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_2s_ancestral</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_sde</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_sde_gpu</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_2m</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_2m_sde</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_2m_sde_gpu</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_3m_sde</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dpmpp_3m_sde_gpu</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>ddim</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>uni_pc</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>uni_pc_bh2</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="16" column="1">
        <widget class="QSpinBox" name="iterationsValue">
         <property name="toolTip">
          <string>Number of iterations to perform</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>128</number>
         </property>
        </widget>
       </item>
       <item row="22" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Sampler:</string>
         </property>
        </widget>
       </item>
       <item row="15" column="1">
        <widget class="QSpinBox" name="seedValue">
         <property name="toolTip">
          <string>Seed to use. -1 for random</string>
         </property>
         <property name="minimum">
          <number>-1</number>
         </property>
         <property name="maximum">
          <number>999999999</number>
         </property>
         <property name="value">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QSpinBox" name="imgHeightValue">
         <property name="toolTip">
          <string>Height of the generated image</string>
         </property>
         <property name="minimum">
          <number>256</number>
         </property>
         <property name="maximum">
          <number>4096</number>
         </property>
         <property name="singleStep">
          <number>64</number>
         </property>
         <property name="value">
          <number>512</number>
         </property>
        </widget>
       </item>
       <item row="30" column="1" colspan="2">
        <widget class="QComboBox" name="sdxlRefinerCheckpointCombo"/>
       </item>
       <item row="17" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Batch size:</string>
         </property>
        </widget>
       </item>
       <item row="19" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Steps:</string>
         </property>
        </widget>
       </item>
       <item row="18" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>CFG scale:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Model:</string>
         </property>
        </widget>
       </item>
       <item row="18" column="1">
        <widget class="QSpinBox" name="cfgValue">
         <property name="toolTip">
          <string>CFG scale value</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>40</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item row="17" column="1">
        <widget class="QSpinBox" name="batchValue">
         <property name="toolTip">
          <string>How many images to generate at once. More VRAM</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>64</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="15" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Seed:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="loraTab">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>LoRA</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>LoRA model:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QDoubleSpinBox" name="loraStrengthSpin">
         <property name="toolTip">
          <string>Select the model strength of the LoRA</string>
         </property>
         <property name="maximum">
          <double>2.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.050000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="Line" name="line_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="loraCheck">
         <property name="toolTip">
          <string>Enable a LoRA when generating images</string>
         </property>
         <property name="text">
          <string>Enable LoRA</string>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QDoubleSpinBox" name="loraClipStrengthSpin">
         <property name="toolTip">
          <string>Select the clip strength of the LoRA</string>
         </property>
         <property name="maximum">
          <double>2.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.050000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QComboBox" name="loraCombo">
         <property name="toolTip">
          <string>Select SD1/2 LoRA</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>LoRA model strength:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>LoRA clip strength:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="modelsTab">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>Paths</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>ComfyUI extra checkpoints folder:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="reloadModelsButton">
         <property name="toolTip">
          <string>Reload models</string>
         </property>
         <property name="text">
          <string>Reload</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="comfyuiExtraFolderValue">
         <property name="toolTip">
          <string>Path to extra checkpoints folder, must be recognised by ComfyUI</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>ComfyUI models folder: </string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QToolButton" name="comfyuiModelFolderSelect">
         <property name="toolTip">
          <string>Select the 'ComfyUI/models' folder</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QToolButton" name="comfyuiExtraFolderSelect">
         <property name="toolTip">
          <string>Select a ComfyUI extra models folder</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="comfyuiModelFolderValue">
         <property name="toolTip">
          <string>Path to the 'ComfyUI/models' folder</string>
         </property>
         <property name="text">
          <string>ComfyUI/models</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Preferences</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Max images:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="keepImagesCheck">
         <property name="text">
          <string>Keep images in display</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="maxImagesSpin">
         <property name="minimum">
          <number>32</number>
         </property>
         <property name="maximum">
          <number>128</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>useExternalVaeCheck</tabstop>
  <tabstop>imgWidthValue</tabstop>
  <tabstop>imgHeightValue</tabstop>
  <tabstop>seedValue</tabstop>
  <tabstop>iterationsValue</tabstop>
  <tabstop>batchValue</tabstop>
  <tabstop>cfgValue</tabstop>
  <tabstop>stepsValue</tabstop>
  <tabstop>loraCheck</tabstop>
  <tabstop>loraCombo</tabstop>
  <tabstop>loraStrengthSpin</tabstop>
  <tabstop>loraClipStrengthSpin</tabstop>
  <tabstop>comfyuiModelFolderValue</tabstop>
  <tabstop>comfyuiModelFolderSelect</tabstop>
  <tabstop>comfyuiExtraFolderValue</tabstop>
  <tabstop>comfyuiExtraFolderSelect</tabstop>
  <tabstop>saveSettingsButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
